### DISCUSSION TITLE
Which is the pixel format of the yuv output?

### DISCUSSION THREAD

**Discussion by rafael2k** [fraunhoferhhi/vvdec#50] [Q&A]
Which is the pixel format of the yuv output?
I'm trying to convert with ffmpeg to a format I can read, but I don't know which format to choose (yuv422p did not work, yuv420p obviously not too).

**Comment by rafael2k** [comment]
Answering myself... yuv420p10le

**Comment by moha23** [comment]
Hi @rafael2k, thanks for this. My input yuv is 420p 8bit, encoded with vvenc to get a .266 bitstream, then decoding even with the VTM decoder software to a yuv file and extracting the frames with ffmpeg gave wrong results. Eg. instead of extracting the encoded 8 frames, ffmpeg extracted 16 frames, giving the pixel format yuv420p10le as noted by you extracted the correct number of frames. So this is the default pixel format they are encoding the input yuv to.. I think?

**Comment by adamjw24** [comment]
Dear moha23,
so the output format of the decoder is defined by the bitstream. For 8 bit input bitstream, the decoder output is yuv420p, for 10 bit input bitstream, the decoder output is yuv420p10le. The idea behind that is, the decoder just translates the bitstream into its exact representation as a video. Any reformatting would be a post-processing step, i.e. not part of the actual decoder.
For the VVenC encoding, the default is to encode everything with 10-bit internal bit-depth. For the expert app this can be set using the InternalBitDepth parameter. For the simple app, we will add an appropriate option for a new release coming up next week.
Hope this helps resolve the confusion.

**Comment by moha23** [comment]
makes sense, thank you for clarifying!

**Comment by picagrad** [comment]
Did this ever get added to the simple app? Trying to encode a picture with vvencapp, decode with vvencapp, and then convert back to jpg/png for processing with some vision software that can only use those formats. Ideally the decoded yuv will have the same bit-depth as the input (which is a simple 8-bit yuv420)

**Comment by lehmann-c** [comment]
Yes it´s available in the simple app:
        `vvencapp --fullhelp `
        `--internal-bitdepth [10]     internal bitdepth (8, 10)`
Use `--internal-bitdepth 8` and the video is encoded in 8bit and the decoded output will be 8 bit yuv420
